task/control-plane-agent/src/mgs_gimlet.rs

```diff
@@ -908,6 +917,46 @@ impl SpHandler for MgsHandler {
             .map_err(|_| SpError::Busy)?;
         Ok(())
     }
+
+    fn set_ipcc_key_lookup_value(
+        &mut self,
+        _sender: SocketAddrV6,
+        _port: SpPort,
+        key: u8,
+        value: &[u8],
+    ) -> Result<(), SpError> {
+        use gateway_messages::IpccKeyLookupValueError;
+        use host_sp_messages::Key;
+
+        ringbuf_entry!(Log::MgsMessage(MgsMessage::SetIpccKeyValue {
+            key,
+            value_len: value.len(),
+        }));
+
+        match Key::from_u8(key) {
+            Some(Key::InstallinatorImageId) => {
+                if value.len() > self.installinator_image_id.capacity() {
+                    return Err(SpError::SetIpccKeyLookupValueFailed(
+                        IpccKeyLookupValueError::ValueTooLong {
+                            max_len: self.installinator_image_id.capacity()
+                                as u16,
+                        },
+                    ));
+                }
+
+                self.installinator_image_id.clear();
+                self.installinator_image_id
+                    .extend_from_slice(value)
+                    .unwrap_lite();
```

Comment:
You could also remove the check above entirely and do something like
```
        self.installinator_image_id
            .extend_from_slice(value)
            .ok_or_else(|| {
                let max_len = self.installinator_image_id.capacity();
                SpError::SetIpccKeyLookupValueFailed(
                    IpccKeyLookupValueError::ValueTooLong {
                        max_len: max_len as u16,
                    },
                )
            })?;
```
(dunno if this passes the borrow checker; you might need to save `max_len` beforehand)

Reply:
Yeah, I don't feel too strongly about it either way.

Reply:
Added a couple short comments explaining why it's structured this way in 68473bc3